src/cli.rs

```diff
@@ -109,6 +131,73 @@ pub fn run() -> Result<(), String> {
 
 // Private helper functions
 
+fn get_login_details(args: &CmdArgs) -> Result<LoginDetails, String> {
+    let mut the_secret: String = String::from("");
+    let mut the_password: String = String::from("");
+
+    let environment_details = envy::from_env::<Environment>().unwrap();
+
+    if let Some(safe_auth_secret) = environment_details.safe_auth_secret {
+        the_secret = safe_auth_secret;
+        info!("Using secret from provided ENV var: safe_auth_secret")
+    }
+
+    if let Some(safe_auth_password) = environment_details.safe_auth_password {
+        the_password = safe_auth_password;
+        info!("Using password from provided ENV var: safe_auth_password")
+    }
+
+    if (!the_secret.is_empty() && the_password.is_empty())
```

Comment:
Since you are comparing bools perhaps it's safe to use the bitwise xor operator, e.g.  `if the_secret.is_empty() ^ the_password.is_empty() {` ?